text/0057-general-multiplication.md

```diff
@@ -0,0 +1,91 @@
+# General multiplication in Cedar
+
+## Related issues and PRs
+
+- Reference Issues: related to [RFC 50]
+- Implementation PR(s):
+
+## Timeline
+
+- Started: 2024-03-07
+
+## Summary
+
+Allow multiplication of arbitrary expressions (that evaluate to Long), not just
+multiplication of an expression by a constant.
+
+## Basic example
+
+```
+permit(principal, action, resource) when {
+    context.foo * principal.bar >= 100
+};
+```
+
+## Motivation
+
+Today, Cedar only supports multiplication of two operands if (at least) one
+operand is a constant.
+In the more general n-ary (chained) multiplication case, Cedar requires that at
+most one operand is a non-constant, but allows the constants and non-constant to
+appear in any order.
+
+- This creates some confusion for Cedar users. No other programming language
+  that I know of has this restriction on multiplication.
```

Comment:
Style nit:  this is the only occurrence of first person singular in the RFC.  I'd replace it or drop the sentence.

Reply:
Fixed